FirebaseInstallations/Source/FIRSecureStorage.m

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRSecureStorage.h"
+#import <Security/Security.h>
+
+#import <FBLPromises/FBLPromises.h>
+
+#import "FIRInstallationsErrorUtil.h"
+
+@interface FIRSecureStorage ()
+@property(nonatomic, strong) dispatch_queue_t keychainQueue;
+@property(nonatomic, readonly) NSString *service;
+@end
+
+@implementation FIRSecureStorage
+
+- (instancetype)init {
+  return [self initWithService:@"com.firebase.FIRInstallations.installations"];
+}
+
+- (instancetype)initWithService:(NSString *)service {
+  self = [super init];
+  if (self) {
+    _keychainQueue = dispatch_queue_create("com.firebase.FIRSecureStorage", DISPATCH_QUEUE_SERIAL);
+    _service = [service copy];
+  }
+  return self;
+}
+
+- (FBLPromise<id<NSSecureCoding>> *)getObjectForKey:(NSString *)key
+                                        objectClass:(Class)objectClass
+                                        accessGroup:(nullable NSString *)accessGropup {
+  return [FBLPromise onQueue:self.keychainQueue
+                          do:^id {
+                            NSDictionary *query = [self keychainQueryWithKey:key
+                                                                 accessGroup:accessGropup];
+                            NSError *error;
+                            NSData *encodedObject = [self getItemWithQuery:query error:&error];
+
+                            if (error) {
+                              return error;
+                            }
+
```

Comment:
Too many blank lines?

Reply:
I'll remove some in the next PRs if you don't mind